tests/abi/tools/revng-check-compatibility-with-abi/ABIRuntimeTestResultParser.h

```diff
@@ -4,86 +4,137 @@
 // This file is distributed under the MIT License. See LICENSE.md for details.
 //
 
+#include <optional>
+
+#include "llvm/ADT/ArrayRef.h"
 #include "llvm/ADT/StringRef.h"
 
 #include "revng/ADT/SortedVector.h"
 #include "revng/Model/Architecture.h"
 #include "revng/Model/Register.h"
 
-namespace abi::artifact {
-
-struct Register {
-  llvm::StringRef Name;
-  uint64_t Value;
-  std::vector<std::byte> Bytes;
-};
-using Registers = llvm::SmallVector<Register, 32>;
-using Stack = llvm::SmallVector<std::byte, 32 * 8>;
+namespace abi::runtime_test {
 
 struct Argument {
   llvm::StringRef Type;
   std::vector<std::byte> Bytes;
   uint64_t MaybePointer;
 };
-using Arguments = llvm::SmallVector<Argument, 32>;
 
-struct ModelRegister {
+struct RawWord {
+  llvm::StringRef Name;
+  uint64_t Value;
+  std::vector<std::byte> Bytes;
+};
+
+struct Register {
   model::Register::Values Name;
-  uint64_t Value = 0;
-  std::vector<std::byte> Bytes = {};
+  uint64_t Value;
+  std::vector<std::byte> Bytes;
+
+  Register(model::Register::Values Name,
+           uint64_t Value = 0,
+           std::vector<std::byte> Bytes = {}) :
+    Name(Name), Value(Value), Bytes(Bytes) {}
+
+  Register(RawWord &&Raw, model::Architecture::Values Architecture) :
+    Name(model::Register::fromRegisterName(Raw.Name, Architecture)),
+    Value(Raw.Value),
+    Bytes(std::move(Raw.Bytes)) {}
+
+  Register(const Register &) = default;
+  Register(Register &&) = default;
+  Register &operator=(const Register &) = default;
+  Register &operator=(Register &&) = default;
 };
 
-} // namespace abi::artifact
+} // namespace abi::runtime_test
 
 template<>
-struct KeyedObjectTraits<abi::artifact::ModelRegister> {
+struct KeyedObjectTraits<abi::runtime_test::Register> {
   static model::Register::Values
-  key(const abi::artifact::ModelRegister &Object) {
+  key(const abi::runtime_test::Register &Object) {
     return Object.Name;
   }
-  static abi::artifact::ModelRegister
+  static abi::runtime_test::Register
   fromKey(const model::Register::Values &Key) {
     return { Key };
   }
 };
 
-namespace abi::artifact {
+namespace abi::runtime_test {
+
+struct State {
+  SortedVector<Register> Registers;
```

Comment:
Do not use `SortedVector` in place of a `std::set`. It's not its purpose.

Reply:
Done.